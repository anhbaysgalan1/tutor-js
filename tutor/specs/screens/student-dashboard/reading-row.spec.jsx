import React from 'react';
import SnapShot from 'react-test-renderer';
import { cloneDeep } from 'lodash';
import moment from 'moment-timezone';
import chronokinesis from 'chronokinesis';

import MOCK_DASHBOARD_RESPONSE from '../../../api/courses/1/dashboard';
import ReadingRow from '../../../src/screens/student-dashboard/reading-row';


describe('Reading Row', function() {
  let props;

  beforeEach(function() {
    chronokinesis.travel(new Date('2017-10-14T12:00:00.000Z'));
    moment.tz.setDefault('America/Chicago');
    props = {
      event: cloneDeep(MOCK_DASHBOARD_RESPONSE.tasks[1]),
      courseId: '1',
      isCollege: true,
    };
  });

  it('renders unstarted', function() {
    props.event.complete_exercise_count = 0;
    expect(SnapShot.create(<ReadingRow {...props} />).toJSON()).toMatchSnapshot();
  });

  it('renders in progress', function() {
    props.event.complete_exercise_count = 1;
    expect(SnapShot.create(<ReadingRow {...props} />).toJSON()).toMatchSnapshot();
  });

  it('renders complete', function() {
    props.event.complete = true;
    expect(SnapShot.create(<ReadingRow {...props} />).toJSON()).toMatchSnapshot();
  });

});
