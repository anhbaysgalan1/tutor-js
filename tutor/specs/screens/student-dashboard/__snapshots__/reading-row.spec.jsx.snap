// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reading Row renders complete 1`] = `
<a
  aria-label="Work on reading: viral cultivate applications"
  className="task row reading workable"
  data-event-id={3}
  href="/course/3/task/3"
  onClick={[Function]}
  onKeyDown={undefined}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-xs-2"
  >
    <i
      aria-label="reading icon"
      className="icon icon-lg icon-reading"
    />
  </div>
  <div
    className="title col-sm-6 col-xs-10"
  >
    viral cultivate applications
  </div>
  <div
    className="feedback col-sm-3 col-xs-5"
  >
    <span>
      Complete
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-xs-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;

exports[`Reading Row renders in progress 1`] = `
<a
  aria-label="Work on reading: out-of-the-box embrace supply-chains"
  className="task row reading workable"
  data-event-id={2}
  href="/course/2/task/2"
  onClick={[Function]}
  onKeyDown={undefined}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-xs-2"
  >
    <i
      aria-label="reading icon"
      className="icon icon-lg icon-reading"
    />
  </div>
  <div
    className="title col-sm-6 col-xs-10"
  >
    out-of-the-box embrace supply-chains
  </div>
  <div
    className="feedback col-sm-3 col-xs-5"
  >
    <span>
      In progress
    </span>
    <span
      className="tour-anchor"
      data-tour-anchor-id="about-late-icon"
      id="about-late-icon"
    >
      <i
        className="info late"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      />
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-xs-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;

exports[`Reading Row renders unstarted 1`] = `
<a
  aria-label="Work on reading: turn-key leverage content"
  className="task row reading workable"
  data-event-id={1}
  href="/course/1/task/1"
  onClick={[Function]}
  onKeyDown={undefined}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-xs-2"
  >
    <i
      aria-label="reading icon"
      className="icon icon-lg icon-reading"
    />
  </div>
  <div
    className="title col-sm-6 col-xs-10"
  >
    turn-key leverage content
  </div>
  <div
    className="feedback col-sm-3 col-xs-5"
  >
    <span>
      Complete
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-xs-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;
